spatialmath/mesh.go

```diff
@@ -419,3 +397,135 @@ func (m *Mesh) ToPoints(density float64) []r3.Vector {
 func (m *Mesh) MarshalJSON() ([]byte, error) {
 	return nil, errors.New("MarshalJSON not yet implemented for Mesh")
 }
+
+// boxTriangleIntersectionArea calculates the area of intersection between a box and a triangle.
+// Returns 0 if there's no intersection, the full triangle area if fully enclosed,
+// or the actual intersection area otherwise.
+func boxTriangleIntersectionArea(b *box, t *Triangle) float64 {
+	// Quick check if they don't intersect at all
+	//nolint:errcheck
+	collides, _ := b.CollidesWith(NewMesh(NewZeroPose(), []*Triangle{t}, ""), defaultCollisionBufferMM)
```

Comment:
I want to make sure that this taken care of:

Why is it safe to assume that `t` the triangle does not need to be transformed by some pose, i.e. why is it safe to create a mesh with a ZeroPose.
Suppose we have a mesh which is get from the intel realsense camera.
The mesh's triangles _exist in the frame of the camera_, not in in the world frame.
Suppose we then want to check if a triangle of the mesh collides with a box that exists in the world frame.

So if we just pass in the triangle into this function and do not compose it with the mesh's pose all of this is moot.
My suggestion is that we do
```
func boxTriangleIntersectionArea(b *box, t *Triangle, trianglePose spatialmath.Pose) float64 {
```

Reply:
Yes I just haven't gotten around to this yet

Reply:
Wait hold on - doesn't the triangle get passed in with its own pose in the world frame in the helper function?

I want to make sure we're not double transforming them

Reply:
so, a mesh is a pose + a slice of triangles.
The triangle is composed of 3 vectors. These vectors are it the frame of the camera. To be in the world they must be transformed by the meshes pose.

We can do that before we enter this function, but I would prefer if we did that within this function to be explicit about what is happening.

Reply:
OK getting to this now.  I think there is some overthinking going on.  This is just a helper function to calculate intersection area between a box and a triangle.  There should be consideration for transforming anything, as this is not the correct scope.  Both triangle and box should have a position that is determined by the calling code

Reply:
>  Both triangle and box should have a position that is determined by the calling code

Good, this is what I  would expect from this function.

Reply:
For reference, miko, [here](https://github.com/viamrobotics/sanding/blob/e10367c3ab3a186a1acddf7a845d47794645a568/coverage/coverage_v2.go#L393) is where we use the functions from this branch. Notice we're already getting up the mesh with a transform from the source frame (camera), and need to check against the framesystem state with inputs that I believe will mainly come from [here](https://github.com/viamrobotics/sanding/blob/e10367c3ab3a186a1acddf7a845d47794645a568/coverage/coverage_v2.go#L190)

Reply:
Lets add a comment to `MeshBoxIntersectionArea` which is the only called of this function. 
Within it we do `totalArea += boxTriangleIntersectionArea(b, tri.Transform(m.pose))`.
Can we please add a comment to this line to avoid any confusion down the road.

Reply:
I'm not sure what this response means.  

I feel like this function is perfectly clear.  It calculats the intersection area of a triangle and a box.  Working on providing some tests for this

Reply:
>I'm not sure what this response means.

adding a comment to the function which calls `boxTriangleIntersectionArea` to state the importance/relevance of transforming by the mesh's pose, i.e. a person who is interacting with this code for the first time might not know why this needs to happen.

Reply:
Added a comment